diff --git a/cli/vgmstream_cli_utils.c b/cli/vgmstream_cli_utils.c
index 0c928368..4cb8878e 100644
--- a/cli/vgmstream_cli_utils.c
+++ b/cli/vgmstream_cli_utils.c
@@ -21,6 +21,7 @@ static void clean_filename(char* dst, int clean_paths) {
 void replace_filename(char* dst, size_t dstsize, cli_config_t* cfg, libvgmstream_t* vgmstream) {
     int subsong;
     char stream_name[CLI_PATH_LIMIT];
+    char basename[CLI_PATH_LIMIT];
     char buf[CLI_PATH_LIMIT];
     char tmp[CLI_PATH_LIMIT];
 
@@ -64,6 +65,29 @@ void replace_filename(char* dst, size_t dstsize, cli_config_t* cfg, libvgmstream
             pos[1] = 's'; /* use %s */
             snprintf(tmp, sizeof(tmp), buf, cfg->infilename);
         }
+        else if (pos[1] == 'b') {
+            const char *p_start, *p_end;
+
+            /* get basename: filename without extension */
+            p_start = strrchr(cfg->infilename, '/');
+            if (!p_start) p_start = strrchr(cfg->infilename, '\\');
+            if (p_start) p_start += 1;
+            else p_start = cfg->infilename;
+
+            /* copy filename only */
+            strncpy(basename, p_start, sizeof(basename));
+            basename[sizeof(basename)-1] = '\0';
+
+            /* remove extension */
+            p_end = strrchr(basename, '.');
+            if (p_end != NULL && p_end != basename) {
+                basename[p_end - basename] = '\0';
+            }
+
+            pos[0] = '%';
+            pos[1] = 's'; /* use %s */
+            snprintf(tmp, sizeof(tmp), buf, basename);
+        }
         else if (pos[1] == 's') {
             pos[0] = '%';
             pos[1] = 'i'; /* use %i */
